### PR TITLE
Login feedback if user do not have offline access

### PR DESCRIPTION
**Pull request by carlitorweb**
Pull Request for Issue #39559

I used the core string `JLIB_LOGIN_DENIED = Your access has been denied`. But I think we should create a new one just for this message

### Summary of Changes
`PlgAuthenticationJoomla ` verified is a valid user who try to access, but not check the permission of the offline mode.

### Testing Instructions
Set your site in offline mode. Go to the frontend and try login with a user who have the permission **Offline Access = Not allowed**

### Actual result BEFORE applying this Pull Request
The user cant login, but not message show up of why cant login

### Expected result AFTER applying this Pull Request
The user cant login, but a message show up of why cant login

### CODE REVIEW

**Comment by richard67** [comment]
@carlitorweb As the check added by this PR is in the `else` part of an `$this->app->isClient('administrator')` check, it doesn't only apply to the site but also to other non-admin clients like api and cli. Is that intended?

**Comment by carlitorweb** [comment]
@richard67 For API dont matter. Right now, an unique Bearer token or credentials need be provided in the request. The later have his own separate routine for authenticate the user `Joomla\Plugin\ApiAuthentication\Basic\Extension\Basic` _(by the way is almos the same as PlgAuthenticationJoomla)_

I did not interact much with the CLI yet, but I tested when offline the site I could add a new user without problem.

**Comment by carlitorweb** [comment]
@richard67 can you run the drone again, not sure why fail

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 0f48b71ce46659fc38d08b0f662e616273bf636f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39561">issues.joomla.org/tracker/joomla-cms/39561</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 0f48b71ce46659fc38d08b0f662e616273bf636f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39561">issues.joomla.org/tracker/joomla-cms/39561</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39561">issues.joomla.org/tracker/joomla-cms/39561</a>.</sub>

File: plugins/authentication/joomla/joomla.php [hunk 1]
```diff
@@ -87,14 +87,24 @@ public function onUserAuthenticate($credentials, $options, &$response)
...
+                // Set default status response to success
+                $_status       = Authentication::STATUS_SUCCESS;
+                $_errorMessage = '';
+
                 if ($this->app->isClient('administrator')) {
                     $response->language = $user->getParam('admin_language');
                 } else {
                     $response->language = $user->getParam('language');
+
+                    if ($this->app->get('offline') == 1 && !$user->authorise('core.login.offline')) {
```
Comment (fancyFranci) [inline plugins/authentication/joomla/joomla.php]: The `== 1` is not needed, isn't it?

Comment (roland-d) [reply plugins/authentication/joomla/joomla.php]: @carlitorweb Ping to see the comment :)

Comment (carlitorweb) [reply plugins/authentication/joomla/joomla.php]: @fancyFranci @roland-d thank you

File: plugins/authentication/joomla/joomla.php [hunk 2]
```diff
@@ -83,18 +82,28 @@ public function onUserAuthenticate($credentials, $options, &$response)
 
             if ($match === true) {
                 // Bring this in line with the rest of the system
-                $user               = User::getInstance($result->id);
+                $user    = User::getInstance($result->id);
```
Comment (Quy) [inline plugins/authentication/joomla/joomla.php]: Please revert. Thanks.

Comment (carlitorweb) [reply plugins/authentication/joomla/joomla.php]: I need check what rule is making this mess, is not first time

**Comment by fancyFranci** [comment]
Thank you for making it easier for the users!

**Pull request merged**